### PR TITLE
fix: expose grouped expert LoRA adapter sharing

### PR DESCRIPTION
**Pull request by romitjain-sarvam**
# What does this PR do?

**Expose Megatron-Bridge's `share_expert_adapters` LoRA option through NeMo RL's Megatron PEFT configuration.**

Megatron-Bridge supports two adapter layouts for grouped MoE expert layers:

- `share_expert_adapters: true` shares one adapter across all local experts on an expert-parallel rank.
- `share_expert_adapters: false` creates a separate adapter for each local expert.

NeMo RL previously constructed the Megatron-Bridge `LoRA` configuration without exposing this option. As a result, users could not select per-expert adapters through NeMo RL, and Megatron-Bridge always used its default shared-adapter layout.

This change:

- Adds `share_expert_adapters` as an optional `MegatronPeftConfig` field.
- Applies the configured value to both the training policy and KL reference policy.
- Preserves backward compatibility by leaving the field unset when it is absent, allowing Megatron-Bridge to retain its existing `true` default.

# Issues

NA

# Usage

Set `share_expert_adapters: false` to create one LoRA adapter per local expert:

```yaml
policy:
  megatron_cfg:
    peft:
      enabled: true
      share_expert_adapters: false
```

Existing configurations that omit the field continue using Megatron-Bridge's default shared-adapter behavior.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [x] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information

Local validation completed:

- Ruff formatting and lint checks passed.
- Modified Python files compile under Python 3.13.
- The final commit has a valid DCO sign-off.
- The branch is based directly on the current upstream `main`.

### CODE REVIEW

**Comment by romitjain-sarvam** [comment]
Validate this on a test run, with `Qwen3-30B-A3B`, TP=2, `moe_grouped_gemm=true`, and `share_expert_adapters=false` on commit `14241027472f0dabc7f6238e1b07b66a380226c1`
    
- Confirmed 96 `GroupedExpertLinearAdapter` modules (1 each layer) with 128 local experts each. I used EP=1 so expected to get [128,8,dim] LoRA matrices (For A,B).
- Saved the final model and distributed optimizer checkpoint and resumed from it as well.

```
uv run --locked python examples/run_dpo.py \
  --config examples/configs/dpo.yaml \
  policy.model_name=Qwen/Qwen3-30B-A3B \
  policy.tokenizer.name=Qwen/Qwen3-30B-A3B \
  dpo.max_num_steps=10 \
  dpo.max_num_epochs=1 \
  dpo.val_at_start=false \
  dpo.val_at_end=false \
  dpo.val_period=0 \
  data.shuffle=false \
  policy.max_total_sequence_length=512 \
  policy.train_global_batch_size=2 \
  policy.train_micro_batch_size=1 \
  policy.dtensor_cfg.enabled=false \
  policy.megatron_cfg.enabled=true \
  policy.megatron_cfg.tensor_model_parallel_size=2 \
  policy.megatron_cfg.sequence_parallel=true \
  policy.megatron_cfg.activation_checkpointing=true \
  +policy.megatron_cfg.attention_backend=unfused \
  policy.megatron_cfg.freeze_moe_router=true \
  policy.megatron_cfg.moe_router_load_balancing_type=none \
  policy.megatron_cfg.moe_permute_fusion=true \
  +policy.megatron_cfg.moe_grouped_gemm=true \
  policy.megatron_cfg.peft.enabled=true \
  'policy.megatron_cfg.peft.target_modules=[linear_qkv,linear_fc1,linear_fc2]' \
  policy.megatron_cfg.peft.dim=8 \
  policy.megatron_cfg.peft.alpha=8 \
  +policy.megatron_cfg.peft.share_expert_adapters=false \
  checkpointing.enabled=true \
  checkpointing.save_period=10 \
  checkpointing.save_optimizer=true \
  checkpointing.checkpoint_dir=results/pr4234/checkpoints \
  logger.wandb_enabled=false \
  logger.tensorboard_enabled=false \
  logger.log_dir=results/pr4234/logs \
  cluster.num_nodes=1 \
  cluster.gpus_per_node=2
```

**Review by terrykong — COMMENTED** [review]
@cuichenx could you help review?

**Review by cuichenx — APPROVED** [review]
Changes are correct according to MB's docs and code

**Review by terrykong — APPROVED** [review]
